internal/runners/packages/install.go

```diff
@@ -42,36 +46,58 @@ func (a *Install) Run(params InstallRunParams, nsType model.NamespaceType) (rerr
 		nsTypeV = &nsType
 	}
 
-	ts := params.Timestamp.Time
-	if ts == nil {
+	ts, err := getTime(&params.Timestamp, a.prime.Auth(), a.prime.Project())
+	if err != nil {
+		return errs.Wrap(err, "Unable to get timestamp from params")
+	}
+
+	return requirements.NewRequirementOperation(a.prime).ExecuteRequirementOperation(
+		params.Package.Name,
+		params.Package.Version,
+		params.Revision.Int,
+		0, // bit-width placeholder that does not apply here
+		bpModel.OperationAdded,
+		ns,
+		nsTypeV,
+		ts,
+	)
+}
+
+// getTime returns a timestamp based on the given "--ts" value.
+// If "now" was given, returns "now" according to the platform.
+// If no timestamp was given but the current project's commit time is after the latest inventory
+// timestamp, returns that commit time.
+// Otherwise, returns the specified timestamp or nil (which falls back on the default Platform
+// timestamp for a given operation)
+func getTime(ts *captain.TimeValue, auth *authentication.Auth, proj *project.Project) (*time.Time, error) {
```

Comment:
Please move this into its own file. It's used by `info.go`, `install.go`, `search.go`. Each of those files are specific to their own command.